docs/upgrade-notes/upgrade-task-file-schemas.md

```diff
@@ -0,0 +1,174 @@
+
+Thus, when performing bulk granule operations which pass granules from the database into a Cumulus workflow, it was possible for there to be schema validation failures depending on which task was used to start the workflow and its particular schema.
+
+In order to rectify this situation, [CUMULUS-2388](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2388) was filed and addressed to create a common granule files schema between nearly all of the Cumulus tasks (exceptions discussed below). The following documentation explains the manual changes you need to make to your deployment in order to be compatible with the updated files schema.
+
+## Updated files schema
+
+The updated granule files schema [can be found here](https://github.com/nasa/cumulus/blob/master/packages/schemas/files.schema.json).
+
+These former properties were deprecated (with notes about how to derive the same information from the updated schema, if possible):
+
+- `filename` - concatenate the `bucket` and `key` values with a directory separator (`/`)
+- `name` - use `fileName` property
+- `etag` - ETags are no longer provided as an individual file property. Instead, a separate `etags` object mapping S3 URIs to ETag values is provided as output from the following workflow tasks (guidance on how to integrate this output with your workflows is provided in the [Upgrading your workflows](#upgrading-your-workflows) section below):
+  - `update-granules-cmr-metadata-file-links`
+  - `hyrax-metadata-updates`
+- `fileStagingDir` - no longer supported
+- `url_path` - no longer supported
+- `duplicate_found` - This property is no longer supported, however `sync-granule` and `move-granules` now produce a separate `granuleDuplicates` object as part of their output. The `granuleDuplicates` object is a map of granules by granule ID which includes the files that encountered duplicates during processing. **Please note** that the `granuleDuplicates` output is purely **informational** and does not have any bearing on the separate configuration for how duplicates should be handled. Guidance on how to integrate `granuleDuplicates` information into your workflow configuration is provided below.
+
+## Exceptions
+
+Three workflow tasks did not have their schema for granule files updated:
+
+- `discover-granules`
+- `queue-granules`
+- `parse-pdr`
+
+The reason that these task schemas were not updated is that all of these tasks occur prior to files actually being ingested to S3, thus much of the information that is required in the updated files schema like `bucket`, `key`, or `checksum` is not yet known.
+
+## Upgrading your deployment
+
+### Upgrading your workflows
+
+For any workflows using the `update-granules-cmr-metadata-file-links` task, update the `Parameters` definition for the step like so:
```

Comment:
It might be too complicated, but should we caveat this with *and uses post-to-cmr* or something like 'and uses a downstream task like post-to-cmr that requires etags'? 

Mostly looking to avoid unneeded updates.

Reply:
It wouldn't just be an workflow that has `post-to-cmr` downstream. It would also be if you had `hyrax-metadata-updates` downstream, since that task also accepts `etags` as config

So yeah, it starts to feel a bit complicated to explain. And I'm betting 90% of users just have all three tasks in the same workflow definition. But I'm open to suggestions

Reply:
*nods* that works!